x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java

```diff
@@ -0,0 +1,584 @@
+            SearchListenerPlugin.class,
+            AsyncSearch.class,
+            AsyncResultsIndexPlugin.class,
+            LocalStateCompositeXPackPlugin.class,
+            TestQueryBuilderPlugin.class
+        );
+        return Stream.concat(super.nodePlugins(clusterAlias).stream(), plugs.stream()).collect(Collectors.toList());
+    }
+
+    public static class TestQueryBuilderPlugin extends Plugin implements SearchPlugin {
+        public TestQueryBuilderPlugin() {}
+
+        @Override
+        public List<QuerySpec<?>> getQueries() {
+            QuerySpec<SlowRunningQueryBuilder> slowRunningSpec = new QuerySpec<>(
+                SlowRunningQueryBuilder.NAME,
+                SlowRunningQueryBuilder::new,
+                p -> {
+                    throw new IllegalStateException("not implemented");
+                }
+            );
+
+            return List.of(slowRunningSpec);
+        }
+    }
+
+    public void testCancelViaTasksAPI() throws Exception {
+        setupTwoClusters();
+
+        SearchListenerPlugin.blockQueryPhase();
+
+        SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest("demo", REMOTE_CLUSTER + ":prod");
+        request.setCcsMinimizeRoundtrips(randomBoolean());
+        request.setWaitForCompletionTimeout(TimeValue.timeValueMillis(1));
+        request.setKeepOnCompletion(true);
+        request.getSearchRequest().allowPartialSearchResults(false);
+        request.getSearchRequest().source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()).size(1000));
+
+        AsyncSearchResponse response = submitAsyncSearch(request);
+        assertNotNull(response.getSearchResponse());
+        assertTrue(response.isRunning());
+
+        SearchListenerPlugin.waitSearchStarted();
+
+        ActionFuture<CancelTasksResponse> cancelFuture;
+        try {
+            ListTasksResponse listTasksResponse = client(LOCAL_CLUSTER).admin()
+                .cluster()
+                .prepareListTasks()
+                .setActions(SearchAction.INSTANCE.name())
+                .get();
+            List<TaskInfo> tasks = listTasksResponse.getTasks();
+            assertThat(tasks.size(), equalTo(1));
+            final TaskInfo rootTask = tasks.get(0);
+
+            AtomicReference<List<TaskInfo>> remoteClusterSearchTasks = new AtomicReference<>();
+            assertBusy(() -> {
+                List<TaskInfo> remoteSearchTasks = client(REMOTE_CLUSTER).admin()
+                    .cluster()
+                    .prepareListTasks()
+                    .get()
+                    .getTasks()
+                    .stream()
+                    .filter(t -> t.action().contains(SearchAction.NAME))
+                    .collect(Collectors.toList());
+                assertThat(remoteSearchTasks.size(), greaterThan(0));
+                remoteClusterSearchTasks.set(remoteSearchTasks);
+            });
+
+            for (TaskInfo taskInfo : remoteClusterSearchTasks.get()) {
```

Comment:
should we check that the list is not empty, or better that there is one item per cluster?

Reply:
I think that's handled in the assertBusy: https://github.com/elastic/elasticsearch/pull/97029/files#diff-54ddb375db8a4150e4c6c453be159da93d883315d373d31c2a14d0a2baaccac9R163

It waits until the size is not zero.

In terms of checking for a size of 1, on the remote cluster there is often more than 1:

```
{
  "node" : "DA3ivEjkS76A4emXH3o2Wg",
  "id" : 80,
  "type" : "transport",
  "action" : "indices:data/read/search[phase/query]",
  "start_time" : "2023-06-30T13:51:16.978Z",
  "start_time_in_millis" : 1688133076978,
  "running_time" : "79.8ms",
  "running_time_in_nanos" : 79874042,
  "cancellable" : true,
  "cancelled" : false,
  "parent_task_id" : "msazZKLYS0apvopKa7On0A:66",
  "headers" : { }
}
{
  "node" : "msazZKLYS0apvopKa7On0A",
  "id" : 66,
  "type" : "transport",
  "action" : "internal:transport/proxy/indices:data/read/search[phase/query]",
  "start_time" : "2023-06-30T13:51:16.977Z",
  "start_time_in_millis" : 1688133076977,
  "running_time" : "80.9ms",
  "running_time_in_nanos" : 80931416,
  "cancellable" : true,
  "cancelled" : false,
  "parent_task_id" : "K9DpCDRmRl2ajqALS4DNig:83",
  "headers" : { }
}
```

So I put it into a loop to assert that however many there are, they are not yet cancelled, which is the main concern for the test.